### PR TITLE
fix: Perform the last action with the column name modified.

### PR DESCRIPTION
**Pull request by tashiro-akira**
@AkiraUra @kimusaku 
Fix target column names to revert after removing special characters.
This fix is necessary to address the preprocess issue https://github.com/sapientml/preprocess/issues/8.
Please check.

### CODE REVIEW

File: sapientml_core/generator.py [hunk 1]
```diff
@@ -221,6 +221,29 @@ def generate_pipeline(self, dataset: Dataset, task: Task):
...
+                pipeline.test = (
+                    pipeline.test[:addindex] + "feature_train_csv = feature_train\n    " + pipeline.test[addindex:]
+                )
+                addindex = pipeline.test.index("prediction = pd.DataFrame")
+                pipeline.test = (
+                    pipeline.test[:addindex] + "TARGET_COLUMNS_csv = [TARGET_COLUMNS[0]]\n" + pipeline.test[addindex:]
+                )
+
             pipeline.train = code_block.train + pipeline.train
             pipeline.predict = code_block.predict + pipeline.predict
```
Comment (AkiraUra) [inline sapientml_core/generator.py:248]: How about the following code?

```python
            if "cols_has_symbols" in pipeline.test:
                pipeline.test = pipeline.test.replace('perm_df.to_csv', 'perm_df.rename(columns=rename_symbol_cols).to_csv')
                def replace_targets(match_obj):
                    return match_obj[0].replace('TARGET_COLUMNS', "[rename_symbol_cols.get(v, v) for v in TARGET_COLUMNS]")
                pat = r"prediction = pd.DataFrame\(y_prob, columns=.?TARGET_COLUMNS.*, index=feature_test.index\)"
                pipeline.test = pat.sub(replace_targets, pipeline.test)
                pipeline.predict = pat.sub(replace_targets, pipeline.predict)
            pipeline.validation = code_block.validation + pipeline.validation
            pipeline.test = code_block.test + pipeline.test
            pipeline.train = code_block.train + pipeline.train
            pipeline.predict = code_block.predict + pipeline.predict
```

File: sapientml_core/templates/other_templates/permutation_importance.py.jinja
Comment (AkiraUra) [inline sapientml_core/templates/other_templates/permutation_importance.py.jinja:1]: I don't think these changes are necessary.

File: sapientml_core/templates/other_templates/prediction_result.py.jinja
Comment (AkiraUra) [inline sapientml_core/templates/other_templates/prediction_result.py.jinja:1]: I don't think these changes are necessary.

**Review by AkiraUra — CHANGES_REQUESTED** [review]
Could you consider my comments?

**Comment by tashiro-akira** [comment]
Reflected the content of the review.
Undid unnecessary modifications.

**Review by AkiraUra — COMMENTED** [review]
Thank you for the PR. The code of `rename_dict` is duplicated from the preprocess one.
Could you consider a way of removing the duplication?

**Comment by tashiro-akira** [comment]
Changed branches and issued new pull requests to capture core and preprocess integration fixes.
PR:https://github.com/sapientml/core/pull/78

**Comment by tashiro-akira** [comment]
Changed branches and issued new pull requests to capture core and preprocess integration fixes.

The correspondence of the comment is also described in the new pull request.
PR:https://github.com/sapientml/core/pull/78

**Pull request closed without merging**